src/databricks/labs/ucx/azure/access.py

```diff
@@ -151,16 +156,16 @@ def save_spn_permissions(self) -> str | None:
     def _update_cluster_policy_definition(
         self,
         policy_definition: str,
+        principal_client_id: str,
+        principal_secret_identifier: str,
         storage_accounts: list[StorageAccount],
-        uber_principal: PrincipalSecret,
```

Comment:
another way could be extending the `PrincipalSecret` class and add secret scope props to it, but this way is fine

Reply:
yes, could be confusing as the PrincipalSecret is an Azure resource, where the secret scope and identifier are Databricks resources, but they belong together